sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -368,19 +370,57 @@ private bool TryAddOutputBindingFromMethod(IList<ExpandoObject> bindingMetadata,
 
         private void AddInputTriggerBindingsAndExtensions(IList<ExpandoObject> bindingMetadata, MethodDefinition method)
         {
+            var inputBindingContainsExpression = InputParameterPropertyContainsExpression(method.Parameters);
+
             foreach (ParameterDefinition parameter in method.Parameters)
             {
                 foreach (CustomAttribute parameterAttribute in parameter.CustomAttributes)
                 {
                     if (IsFunctionBindingType(parameterAttribute))
                     {
-                        AddBindingMetadata(bindingMetadata, parameterAttribute, parameter.ParameterType, parameter.Name);
-                        AddExtensionInfo(_extensions, parameterAttribute);
+                        string bindingType = GetBindingType(parameterAttribute);
+                        AddExtensionInfo(_extensions, parameterAttribute, out bool supportsReferenceType);
+
+                        // if this is a trigger and any input binding contains an expression, don't use BindingData
```

Comment:
We need defer bind if any input OR output bindings have binding expressions, right? For example, otherwise, if we were to defer bind for the trigger, the host may not have the binding metadata it needs to resolve the output binding, resulting in a runtime error.

Reply:
That's a good point. I think this class in general is confusing as this current function is called "AddInputTriggerBindingsAndExtensions" and so at this point I assume we only have input and triggers, but there's nothing upstream making that split between input/trigeer and output. I might need to get some clarity from @fabiocav and @brettsam about this, but in general agree with your point about triggers deferring binding if outputs also contain expressions